solution/solution.js

```diff
@@ -2,8 +2,8 @@ const http = require("http");
 const https = require("https");
 
 const myRequest = (url, cb) => {
-  const urlObj = new URL(url);
-  const protocol = urlObj.protocol === "https:" ? https : http;
+  // the line below is extra - the solution to the bonus section
+  const protocol = url.includes("https") ? https : http;
```

Comment:
Maybe it would be clearer to have a `bonus.solution.js`?

Reply:
I've moved the bonus solution into a new file - also added tests for the solution files